addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -633,85 +701,107 @@ var SnippetEditor = Widget.extend({
      * @param {Event} ev
      * @param {Object} ui
      */
-    _onDragAndDropStop: function (ev, ui) {
+    _onDragAndDropStop: async function (ev, ui) {
         // TODO lot of this is duplicated code of the d&d feature of snippets
         if (!this.dropped) {
             var $el = $.nearest({x: ui.position.left, y: ui.position.top}, '.oe_drop_zone', {container: document.body}).first();
             if ($el.length) {
-                $el.after(this.$target);
+                $el.after(this.$snippetBlock);
                 this.dropped = true;
             }
         }
 
-        this.$editable.find('.oe_drop_zone').droppable('destroy').remove();
+        this.$editable.find('.oe_drop_zone').remove();
 
-        var prev = this.$target.first()[0].previousSibling;
-        var next = this.$target.last()[0].nextSibling;
-        var $parent = this.$target.parent();
+        var prev = this.$snippetBlock.first()[0].previousSibling;
+        var next = this.$snippetBlock.last()[0].nextSibling;
+        var $parent = this.$snippetBlock.parent();
 
         var $clone = this.$editable.find('.oe_drop_clone');
         if (prev === $clone[0]) {
             prev = $clone[0].previousSibling;
+            this.dropped = false;
         } else if (next === $clone[0]) {
             next = $clone[0].nextSibling;
+            this.dropped = false;
         }
-        $clone.after(this.$target);
+        $clone.after(this.$snippetBlock);
         var $from = $clone.parent();
 
         this.$el.removeClass('d-none');
         this.$body.removeClass('move-important');
         $clone.remove();
 
-        if (this.dropped) {
-            this.trigger_up('request_history_undo_record', {$target: this.$target});
+        dom.scrollTo(this.$snippetBlock[0], {extraOffset: 50});
```

Comment:
```js
this.trigger_up('request_history_undo_record', {$target: this.$target});
```
was code that handled a scroll? I'm not sure I follow.

Reply:
... I am commenting the `scrollTo` call where you moved it, pointing out the fact it was somewhere else before, at a location so that the actual scroll was smooth. So moving it here (instead of an history recording like you are pointing out) had to be for a good reason but the only effect I see is that it is not smooth anymore.

Reply:
That doesn't seem to make sense to me either indeed. I'm guessing this is part of @Gorash 's answer to the fact that he removed the smooth scroll thing... I hope he can clarify this for us.